server/redux/reducers/user-reducer.js

```diff
@@ -37,14 +37,16 @@ const removeUser = userId => {
 
 /* --------------- THUNK ACTION CREATORS --------------- */
 
-const createAndEmitUser = socket => {
+const createAndEmitUser = (socket, user) => {
   return dispatch => {
     const userId = socket.id;
-    const user = Map(createUser(userId));
-    dispatch(addUser(user));
-    socket.on('sceneLoad', () => {
-      socket.emit('createUser', user);
-    });
+    const displayName = user.displayName;
+    const newUser = Map(createUser(userId, displayName));
+    dispatch(addUser(newUser));
+    console.log('User created');
+    if (socket.sceneLoaded) {
```

Comment:
Nothing.  If user is created first, then when sceneLoad is emitted, the server will send back createUser then.  Basically this line and line 28 in server/socket.js do the same thing, but solve the race condition.  It doesn't matter which happens first, the createUser emit won't fire until both happen.